bake/compose.go

```diff
@@ -141,16 +141,17 @@ func flatten(in compose.MappingWithEquals) compose.Mapping {
 // xbake Compose build extension provides fields not (yet) available in
 // Compose build specification: https://github.com/compose-spec/compose-spec/blob/master/build.md
 type xbake struct {
-	Tags          stringArray `yaml:"tags,omitempty"`
-	CacheFrom     stringArray `yaml:"cache-from,omitempty"`
-	CacheTo       stringArray `yaml:"cache-to,omitempty"`
-	Secrets       stringArray `yaml:"secret,omitempty"`
-	SSH           stringArray `yaml:"ssh,omitempty"`
-	Platforms     stringArray `yaml:"platforms,omitempty"`
-	Outputs       stringArray `yaml:"output,omitempty"`
-	Pull          *bool       `yaml:"pull,omitempty"`
-	NoCache       *bool       `yaml:"no-cache,omitempty"`
-	NoCacheFilter stringArray `yaml:"no-cache-filter,omitempty"`
+	Tags          stringArray       `yaml:"tags,omitempty"`
+	CacheFrom     stringArray       `yaml:"cache-from,omitempty"`
+	CacheTo       stringArray       `yaml:"cache-to,omitempty"`
+	Secrets       stringArray       `yaml:"secret,omitempty"`
+	SSH           stringArray       `yaml:"ssh,omitempty"`
+	Platforms     stringArray       `yaml:"platforms,omitempty"`
+	Outputs       stringArray       `yaml:"output,omitempty"`
+	Pull          *bool             `yaml:"pull,omitempty"`
+	NoCache       *bool             `yaml:"no-cache,omitempty"`
+	NoCacheFilter stringArray       `yaml:"no-cache-filter,omitempty"`
+	Contexts      map[string]string `yaml:"contexts,omitempty"`
```

Comment:
Not sure if we should call this `stringMap` for consistency? (even though the only reason we have `stringArray` is for the custom `UnmarshalYAML` case with CSV parsing).

Reply:
Good point to be consistent, I'm fine with it.